dts/bindings/video/video-interfaces.yaml

```
@@ -0,0 +1,93 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+# Common fields for video interfaces.
+
+include: base.yaml
+
+properties:
+  bus-type:
+    type: int
+    enum:
+      - 1 # Parallel
+      - 2 # MIPI CSI-2 D-PHY
+      - 3 # MIPI CSI-2 C-PHY
+      - 4 # MIPI CSI-1
```

If interested, it looks like Zephyr supports strings, which still turn into integers when building `devicetree_generated.h` but reduces portability a little bit:
https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/usb/usb-controller.yaml#L19-L22

---

There is also [`DT_ON_BUS()`](https://docs.zephyrproject.org/latest/build/dts/api/api.html#c.DT_ON_BUS) to detect between MIPI and parallel ports, then the sensor can detect on which kind of bus it is.

https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/mipi-dsi/mipi-dsi-device.yaml#L8
https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/mipi-dbi/mipi-dbi-device.yaml#L8
https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/mspi/mspi-device.yaml#8

But current DCMI peripherals are not written on top of MSPI, directly a Video API from the SDK, so this might need a `parallel-device.yaml`. More questions!

---

Here is the API for enums in Zephyr devicetree:
https://docs.zephyrproject.org/latest/build/dts/api/api.html#c.DT_ENUM_IDX

```suggestion
       - "parallel"
       - "mipi-csi-2-d-phy"
       - "mipi-csi-2-c-phy"
       - "mipi-csi-1"
```

Then this can be used from anywhere in the drivers:
```C
DT_ENUM_IDX(node, parallel)         // 0
DT_ENUM_IDX(node, mipi_csi_2_d_phy) // 1
DT_ENUM_IDX(node, mipi_csi_2_c_phy) // 2
DT_ENUM_IDX(node, mipi_csi_1)       // 3
```